iree/turbine/aot/tensor_traits.py

```diff
@@ -17,17 +17,12 @@
 ]
 
 
+@dataclass(frozen=True)
```

Comment:
No, device affinity is not a dataclass. You should only use this annotation when the type is struct like.